src/common/snippets/src/lowered/pass/assign_registers.cpp

```diff
@@ -17,281 +18,128 @@ namespace snippets {
 namespace lowered {
 namespace pass {
 
-void AssignRegisters::set_reg_types(LinearIR& linear_ir) {
-    for (const auto& expr : linear_ir) {
-        const auto op = expr->get_node();
-        if (ov::is_type<op::LoopEnd>(op) ||
-            ov::is_type<ov::op::v0::Result>(op)
-#ifdef SNIPPETS_DEBUG_CAPS
-        || ov::is_type<op::PerfCountBeginBase>(op)
-        || ov::is_type<op::PerfCountEndBase>(op)
-#endif
-        )
-        continue;
-
-        OPENVINO_ASSERT(expr->get_output_count() == op->get_output_size(), "Incorrect count of output port descriptors!");
-        for (size_t i = 0; i < expr->get_output_count(); ++i) {
-            const auto reg_type = m_reg_type_mapper(op->output(i));
-            expr->get_output_port_descriptor(i)->set_reg_type(reg_type);
-            // propogate to consumers
-            for (const auto& consumer : expr->get_output_port_connector(i)->get_consumers()) {
-                consumer.get_descriptor_ptr()->set_reg_type(reg_type);
-            }
-        }
+AssignRegisters::RegMap AssignRegisters::assign_regs_manually(const LinearIR& linear_ir, std::set<Reg>& gpr_pool, std::set<Reg>& vec_pool) {
+    RegMap manually_assigned;
+    for (const auto& param : linear_ir.get_parameters()) {
+        manually_assigned[param->get_output_port_descriptor(0)->get_reg()] = *gpr_pool.begin();
```

Comment:
Could you please add assert with `!gpr_pool.empty()`?

Reply:
Sure, added validation here and in other places inside `assign_regs_manually`